lib/prometheus/client/metric.rb

```diff
@@ -55,6 +55,10 @@ def with_labels(labels)
                        store_settings: @store_settings)
       end
 
+      def init_label_set(labels)
+        @store.set(labels: label_set_for(labels), val: 0)
+      end
```

Comment:
I'm not a fan of having a default implementation and some sub classes overriding it, but this seems to be on par with the `#with_labels` implementation.

Reply:
Yeah, sadly, Summaries and Histograms are doing "magic" to the labels to be able to basically wedge more complex objects into a plain hash we can serialise to disk. 
It's a bit of a hack, admittedly, but I couldn't think of anything better...